### PR TITLE
Replace #if FOO with #if defined(FOO) && FOO

### PR DESCRIPTION
**Pull request by ghost**
Prevents warnings about using undefined macros

### CODE REVIEW

File: src/lib/core/TLVReader.cpp [hunk 1]
```diff
@@ -345,7 +345,7 @@ CHIP_ERROR TLVReader::Get(CharSpan & v) const
     }
 
     v = CharSpan(Uint8::to_const_char(bytes), len);
-#if CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ
+#if defined(CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ) && CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ
```
Comment (andy31415) [inline src/lib/core/TLVReader.cpp:348]: This one is defined by lib/core/CHIPConfig.h (actually buildconfig in gn).

We should never have it undefined. Not having this means that the chip config is old or not properly defined.

**Review by andy31415 — CHANGES_REQUESTED** [review]
I believe we currently ensure these are specifically set - we are intentionally using `#if` to trigger compile errors in case something is not properly defined.

Rather than trying to use ifdef, we should figure out why these are undefined and fix that.

File: src/app/AttributePathParams.h [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
 namespace chip {
 namespace app {
-#if CHIP_CONFIG_ENABLE_READ_CLIENT
+#if defined(CHIP_CONFIG_ENABLE_READ_CLIENT) && CHIP_CONFIG_ENABLE_READ_CLIENT
```
Comment (bzbarsky-apple) [inline src/app/AttributePathParams.h:28]: If this is warning, that means that the right config headers are not being included, no?  The right fix is to include those headers.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Review by bauerschwan — CHANGES_REQUESTED** [review]
I would also highly vote against hiding these essential warnings. It would make the code prone to misconfigurations.

**Comment by andy31415** [comment]
OLD PR with changes requested. Closing

**Pull request closed without merging**